lib/super_digit.rb

```diff
@@ -1,15 +1,20 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - 0(n)
+# Space Complexity - 0(n)
+
 def super_digit(n)
-   
+  return n if n < 10
+  sum = n.to_s.chars.map(&:to_i).sum
+  return super_digit(sum)
 end
-  
 
-# Time Complexity - ?
-# Space Complexity - ?
+
+# Time Complexity - 0(n)
+# Space Complexity - 0(n)
 def refined_super_digit(n, k)
```

Comment:
This goes O( (nk)^2) times, you can do better with dynamic programming O(d^2 + k^2)